### DISCUSSION TITLE
Why does my trained audio sound robotic?

### DISCUSSION THREAD

**Discussion by ABSOLUTE-ZER0** [coqui-ai/TTS#2915] [General Q&A]
I'm getting robotic like voice after 70K steps of training, is it normal? If not what did I do wrong? I am very new to all this so forgive me if I sound dumb. 

Sharing my current setup for more details

I am using the SLR115 Dataset to train my model using VITS. I did the following to preprocess my audio:
- Removed white/static noise
- Trimmed starting and ending
- Normalized the audio

Pasting the code i use for audio preprocessing
```
# For white noise reduction
for i in fileArr:
  rate, data = wavfile.read(f"{folder}{i}")
  reduced_noise = nr.reduce_noise(y=data, sr=rate)
  wavfile.write(f"{folder}{i}", rate, reduced_noise)

# For reducing bit rate, for faster training
for file in fileArr:
  if(file.endswith('.wav')):
    nameSolo = file.rsplit('.', 1)[0]
    data, samplerate = soundfile.read(folder + file)                

  soundfile.write(folder + nameSolo + '.wav', data, samplerate, subtype='PCM_16')

# For normalizing audio
for i in fileArr:
  rawsound = AudioSegment.from_file(f"{folder}{i}", "wav")
  normalizedsound = effects.normalize(rawsound)
  normalizedsound.export(f"{folder}{i}", format="wav")

# For trimming audio
for i in fileArr:
  sound: AudioSegment = AudioSegment.from_file(f"{folder}{i}", "wav")
  stripped: AudioSegment = strip_silence(sound)
  stripped.export(f"{folder}{i}", format="wav")
```

Pasting my VITS training file, which i took from the reciepies

```
import os
from multiprocessing import Process, freeze_support

from trainer import Trainer, TrainerArgs

from TTS.tts.configs.shared_configs import BaseDatasetConfig
from TTS.tts.configs.vits_config import VitsConfig
from TTS.tts.datasets import load_tts_samples
from TTS.tts.models.vits import CharactersConfig, Vits, VitsArgs, VitsAudioConfig
from TTS.tts.utils.text.tokenizer import TTSTokenizer
from TTS.utils.audio import AudioProcessor

def main():
	output_path = os.path.dirname(os.path.abspath(__file__))
	dataset_config = BaseDatasetConfig(
		formatter="common_voice", meta_file_train="bea_neutral.tsv", path="D:\\TTS\\LJSpeech\\VITS"
	)
	audio_config = VitsAudioConfig(
		sample_rate=44100, win_length=1024, hop_length=256, num_mels=80, mel_fmin=0, mel_fmax=None
	)

	config = VitsConfig(
		output_path=output_path,
		run_name="vits_bea_neutral",
		dashboard_logger="tensorboard",
		logger_uri=None,
		audio=audio_config,
		batch_size=4,
		eval_batch_size=4,
		batch_group_size=5,
		num_loader_workers=4,
		print_step=50,
		plot_step=100,
		log_model_step=1000,
		save_step=5000,
		save_n_checkpoints=2,
		save_checkpoints=True,
		target_loss="loss_1",
		print_eval=False,
		use_phonemes=True,
		phonemizer="espeak",
		phoneme_language="en",
		compute_input_seq_cache=True,
		add_blank=True,
		text_cleaner="english_cleaners",
		phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
		start_by_longest=True,
		datasets=[dataset_config],
		cudnn_benchmark=False,
		max_audio_len=44100*15,
		min_audio_len=0.5,
		mixed_precision=False,
		test_sentences=[
			"Just writing some dad jokes for testing",
			"Why don't scientists trust atoms?",
			"Because they make up everything!",
		],
	)

	# INITIALIZE THE AUDIO PROCESSOR
	# Audio processor is used for feature extraction and audio I/O.
	# It mainly serves to the dataloader and the training loggers.
	ap = AudioProcessor.init_from_config(config)

	# INITIALIZE THE TOKENIZER
	# Tokenizer is used to convert text to sequences of token IDs.
	# config is updated with the default characters if not defined in the config.
	tokenizer, config = TTSTokenizer.init_from_config(config)

	# LOAD DATA SAMPLES
	# Each sample is a list of ```[text, audio_file_path, speaker_name]```
	# You can define your custom sample loader returning the list of samples.
	# Or define your custom formatter and pass it to the `load_tts_samples`.
	# Check `TTS.tts.datasets.load_tts_samples` for more details.
	train_samples, eval_samples = load_tts_samples(
		dataset_config,
		eval_split=True,
		eval_split_max_size=config.eval_split_max_size,
		eval_split_size=config.eval_split_size,
	)

	# init model
	model = Vits(config, ap, tokenizer, speaker_manager=None)

	# init the trainer and 🚀
	trainer = Trainer(
		TrainerArgs(),
		config,
		output_path,
		model=model,
		train_samples=train_samples,
		eval_samples=eval_samples,
	)
	trainer.fit()

if __name__ == '__main__':
	freeze_support()  # needed for Windows
	main()
```

I want to know what params like num_mels, win_length, and hop_length do. Can it be the reason why my audio sounds robotic? Even my training voice sounds robotic when I preview it in Pytorch. Adding samples here

**Original audio**
- https://drive.google.com/file/d/1gnpFkWmtlznAZ961t90JIieQBh-p6SDg/view?usp=drive_link

**Preprocessed audio**
- https://drive.google.com/file/d/1eZ5PXKrciQP_YQnxQ-XRF0WV6Nzr4mkg/view?usp=drive_link

**Training audio**
- Step 67K: https://drive.google.com/file/d/1SoeEALok2P_he1f78Y0Ui-RcRBItTvyk/view?usp=drive_link
- Step 70K: https://drive.google.com/file/d/1kIc6iNK18Ut1F7xxeS4P_ySMsAIu0fBL/view?usp=drive_link

**Testing audio**
- Step 67K: https://drive.google.com/file/d/1J8inJHScqHk-YqVhK4fBFUr6MYw0-lIo/view?usp=drive_link
- Step 70K: https://drive.google.com/file/d/1Co-e7ek-cy0gzVxof8gCeZio5EQIn9O6/view?usp=drive_link
